tests/run-pass/aligned_utf8_check.rs

```diff
@@ -0,0 +1,13 @@
+fn main() {
+    const N: usize = 10;
+
+    let x = vec![0x4141u16; N];
+
+    let mut y: Vec<u8> = unsafe { std::mem::transmute(x) };
+    unsafe { y.set_len(2 * N) };
+
+    println!("{:?}", std::str::from_utf8(&y).unwrap());
```

Comment:
The test should come with a comment explaining why this tests anything.

I think this actually doesn't test anything. The UTF-8 validation calls `align_offset(mem::size_of::<usize>())`, so we have `align == 8` and `stride == 2` in the code above, and we still hit the old code path.

Reply:
That's my fault. I misremember what happens

Reply:
So, please change this to a `vec!` of `u64`. That should then actually test something.

Also, instead of transmuting the entire `Vec` around, I think it is much easier and safer to cast a slice around:
```rust
// rename x to vec or so. then:
let content = slice::from_raw_parts(vec.as_ptr(), 8*N);
println!("{:?}", std::str::from_utf8(content).unwrap());
```